presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/JdbcMetadata.java

```diff
@@ -272,12 +278,58 @@ public JdbcTableHandle getTableHandle(ConnectorSession session, SchemaTableName
                                 .map(JdbcColumnHandle.class::cast)
                                 .collect(toImmutableList()))
                         .collect(toImmutableList())),
-                OptionalLong.empty(), // limit
+                handle.getLimit(), // limit
+                handle.getSortOrder(), //orderby
                 Optional.of(newColumns.build()));
 
         return Optional.of(new AggregationApplicationResult<>(handle, projections.build(), resultAssignments.build(), ImmutableMap.of()));
     }
 
+    @Override
+    public Optional<TopNApplicationResult<ConnectorTableHandle>> applyTopN(
+            ConnectorSession session,
+            ConnectorTableHandle table,
+            long topNCount,
+            List<SortItem> sortItems,
+            Map<String, ColumnHandle> assignments)
+    {
+        if (!isAllowTopNPushdown(session) || !jdbcClient.supportsTopNPushDown() || !jdbcClient.supportsLimit()) {
```

Comment:
`jdbcClient.supportsLimit()` should be handled internally by supportsTopNPushDown.